.github/workflows/update_protos.yml

```diff
@@ -0,0 +1,47 @@
+name: Update Protos + Open PR
+
+on:
+  workflow_dispatch:
+  repository_dispatch:
+    types:
+      - protos-updated
+
+jobs:
+  update-protos:
+    if: github.repository_owner == 'viamrobotics'
+    runs-on: [self-hosted, x64]
+    container:
+      image: ghcr.io/cirruslabs/flutter:3.10.6
+    steps:
+      - uses: actions/checkout@v3
+      - uses: bufbuild/buf-setup-action@v1.25.0
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+      - uses: arduino/setup-protoc@v2
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Setup Flutter
+        run: flutter pub get
+
+      - run: make setup
+
+      - name: Generate buf
+        run: make buf
+        env:
+          BUF_TOKEN: ${{ secrets.BUF_TOKEN }}
+
+      - name: Format
+        run: make format
+
+      - name: Add + Commit + Open PR
+        uses: peter-evans/create-pull-request@v5
```

Comment:
Still blows my mind that this is not an action provided by GitHub lol who is peter evans.

Reply:
Right?? I think you can manually do it since actions always include the GH command line tool but still